diff -rupN ZSI/client.py ZSI_patched/client.py
--- ZSI/client.py	2009-05-13 18:28:10.000000000 -0400
+++ ZSI_patched/client.py	2009-05-13 18:09:37.000000000 -0400
@@ -380,6 +380,13 @@ class _Binding:
     def IsSOAP(self):
         if self.ps: return 1
         self.ReceiveRaw()
+        # Is this running on Google's App Engine?
+        try:
+            import google.appengine
+            if self.reply_code == 200:
+                self.reply_headers.type = 'text/xml'
+        except:
+            pass
         mimetype = self.reply_headers.type
         return mimetype == 'text/xml'
 
diff -rupN ZSI/parse.py ZSI_patched/parse.py
--- ZSI/parse.py	2009-05-13 18:28:10.000000000 -0400
+++ ZSI_patched/parse.py	2009-05-13 18:10:33.000000000 -0400
@@ -51,7 +51,12 @@ class ParsedSoap:
             if self.defaultReaderClass != None:
                 self.readerclass = self.defaultReaderClass
             else:
-                from xml.dom.ext.reader import PyExpat
+                # Is this running on Google's App Engine?
+                try:
+                    import google.appengine
+                    from _xmlplus.dom.ext.reader import PyExpat
+                except:
+                    from xml.dom.ext.reader import PyExpat
                 self.readerclass = PyExpat.Reader
         try:
             self.reader = self.readerclass()
@@ -207,7 +212,7 @@ class ParsedSoap:
             if t != _Node.ELEMENT_NODE:
                 if t == _Node.TEXT_NODE and n.nodeValue.strip() == "":
                     continue
-                raise ParseException("Non-element child in " + name, 
+                raise ParseException("Non-element child in " + name,
                         inheader, elt, self.dom)
             if mustqualify and not n.namespaceURI:
                 raise ParseException('Unqualified element "' + \
@@ -347,7 +352,7 @@ class ParsedSoap:
         lenofwhat = len(ofwhat)
         c, crange = self.header_elements[:], range(len(self.header_elements))
         for i,what in [ (i, ofwhat[i]) for i in range(lenofwhat) ]:
-            if isinstance(what, AnyElement): 
+            if isinstance(what, AnyElement):
                 raise EvaluateException, 'not supporting <any> as child of SOAP-ENC:Header'
 
             v = []
